### ISSUE TITLE
eval_mode giving inconsistent results

### ISSUE THREAD

**Issue by rijobro** [Project-MONAI/MONAI#1728]
**Describe the bug**
Some networks give different results each time they're called with `eval_mode`.

I suspect this might be the cause of https://github.com/Project-MONAI/MONAI/issues/1710.

**To Reproduce**
```python
import torch
from monai.networks.utils import eval_mode

model = torch.nn.BatchNorm2d(1)
im = torch.randn((1, 1, 2, 2))
for _ in range(5):
    res_train = model(im).flatten().detach()
    with eval_mode(model):
        res_eval = model(im).flatten()
    print(res_train)
    print(res_eval)
```

Example output, notice how odd rows all match (train), and even rows differ (eval):
```python
tensor([ 0.7991,  0.5358, -1.7123,  0.3774])
tensor([ 0.7824,  0.4933, -1.9753,  0.3193])
tensor([ 0.7991,  0.5358, -1.7123,  0.3774])
tensor([ 0.7689,  0.4882, -1.9086,  0.3193])
tensor([ 0.7991,  0.5358, -1.7123,  0.3774])
tensor([ 0.7579,  0.4842, -1.8531,  0.3195])
tensor([ 0.7991,  0.5358, -1.7123,  0.3774])
tensor([ 0.7489,  0.4810, -1.8064,  0.3198])
tensor([ 0.7991,  0.5358, -1.7123,  0.3774])
tensor([ 0.7413,  0.4784, -1.7667,  0.3201])
```

**Comment by wyli** [comment]
I feel this is expected because `res_train` updates the BN running stats, `res_eval` use the updated values...

**Comment by wyli** [comment]
you're right, the visualisation module shouldn't quietly update the model. shall we set the BN layers to eval() even if the model is in `train_mode`?

**Comment by rijobro** [comment]
I wondered if we should use the reset model that we implemented for the learning rate finder?

**Comment by rijobro** [comment]
Or would it be possible to only enable the gradient of the target layer?

**Comment by wyli** [comment]
I had a quick test it seems we shouldn't use `eval_mode` in the vis, we want `eval()` but without `torch.no_grad()`
